packages/@lwc/template-compiler/src/__tests__/fixtures/attributes/boolean-attribute/hidden-global-attr/expected.js

```diff
@@ -0,0 +1,116 @@
+import _xFoo from "x/foo";
+import { registerTemplate } from "lwc";
+
+function tmpl($api, $cmp, $slotset, $ctx) {
+  const { t: api_text, h: api_element, c: api_custom_element } = $api;
+  return [
+    api_element(
+      "p",
+      {
+        attrs: {
+          hidden: "",
+        },
+        key: 0,
+      },
+      [api_text("boolean present")]
+    ),
+    api_element(
+      "p",
+      {
+        attrs: {
+          hidden: "",
+        },
+        key: 1,
+      },
+      [api_text("empty string, should be true")]
+    ),
+    api_element(
+      "p",
+      {
+        attrs: {
+          hidden: "other than true",
+        },
+        key: 2,
+      },
+      [api_text("string value, should be true")]
+    ),
+    api_element(
+      "p",
+      {
+        attrs: {
+          hidden: $cmp.computed ? "" : null,
+        },
+        key: 3,
+      },
+      [api_text("computed value, should be resolved in component")]
+    ),
+    api_element(
+      "p",
+      {
+        attrs: {
+          hidden: "3",
+        },
+        key: 4,
+      },
+      [api_text("integer value, should be true")]
+    ),
+    api_custom_element(
+      "x-foo",
+      _xFoo,
+      {
+        props: {
+          hidden: true,
+        },
+        key: 5,
+      },
+      [api_text("boolean present")]
+    ),
+    api_custom_element(
+      "x-foo",
+      _xFoo,
+      {
+        props: {
+          hidden: true,
+        },
+        key: 6,
+      },
+      [api_text("empty string, should be true")]
+    ),
+    api_custom_element(
+      "x-foo",
+      _xFoo,
+      {
+        props: {
+          hidden: true,
+        },
+        key: 7,
+      },
+      [api_text("string value, should be true")]
+    ),
+    api_custom_element(
+      "x-foo",
+      _xFoo,
+      {
+        props: {
+          hidden: $cmp.computed,
```

Comment:
Looking at the compiled code I am tempted to normalize the property to a boolean: `hidden: !!$cmp.computed`. The reasoning here is that it would match with how static attributes are converted to properties. For example `hidden="3"` attribute is transformed into `hidden: true` prop.

Any thoughts on this?

Reply:
i would like to leave this as is because today we don't enforce it, therefore could be some component with an api attribute `hidden` that is receiving other than boolean values; normalizing the values will be breaking change for such components. that being said, the probability is low.

Reply:
Ok, no problem. Do you mind opening an issue so we don't forget in the future?